jabgui/src/main/java/org/jabref/gui/frame/ImportHelper.java

```diff
@@ -0,0 +1,98 @@
+package org.jabref.gui.frame;
+
+import java.net.MalformedURLException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import org.jabref.gui.preferences.GuiPreferences;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.ImportException;
+import org.jabref.logic.importer.ImportFormatReader;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.net.URLDownload;
+import org.jabref.logic.os.OS;
+import org.jabref.model.util.DummyFileUpdateMonitor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ImportHelper {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImportHelper.class);
+
+    /**
+     * Reads URIs as input
+     */
+    static Optional<ParserResult> importFile(String importArguments,
+                                             String importFormat,
+                                             GuiPreferences cliPreferences,
+                                             boolean porcelain) {
+        LOGGER.debug("Importing file {}", importArguments);
+        String[] data = importArguments.split(",");
+
+        String address = data[0];
+        Path file;
+        if (address.startsWith("http://") || address.startsWith("https://") || address.startsWith("ftp://")) {
+            // Download web resource to temporary file
+            try {
+                file = new URLDownload(address).toTemporaryFile();
+            } catch (FetcherException |
+                    MalformedURLException e) {
+                System.err.println(Localization.lang("Problem downloading from %1", address) + e.getLocalizedMessage());
+                return Optional.empty();
+            }
+        } else {
+            if (OS.WINDOWS) {
+                file = Path.of(address);
+            } else {
+                file = Path.of(address.replace("~", System.getProperty("user.home")));
+            }
+        }
+
+        Optional<ParserResult> importResult = importFile(file, importFormat, cliPreferences, porcelain);
+        importResult.ifPresent(result -> {
+            if (result.hasWarnings()) {
+                System.out.println(result.getErrorMessage());
+            }
+        });
+        return importResult;
+    }
+
+    static Optional<ParserResult> importFile(Path file,
```

Comment:
The current cli should still have the import available; only renamed.

I checked the browser extension - and it sends bibtex, doesn't it?

Reply:
No, on Windows it sends a temp file (check the jabrefhost.py) and on mac and I think linux it sends bibtex